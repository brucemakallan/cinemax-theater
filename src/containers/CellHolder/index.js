import React, { Component } from 'react';
import Cell from '../../components/Cell';
import PropTypes from 'prop-types';

class CellHolder extends Component {
	handleOnClick = (id) => {
		// console.log('clicked ', id);
	}
	render() { 
		const { value, type, id } = this.props;
		return (
			<Cell value={value} type={type} id={id} onClick={this.handleOnClick} />
		);
	}
}

CellHolder.propTypes = {
	value: PropTypes.string.isRequired,
	type: PropTypes.string.isRequired,
	id: PropTypes.string.isRequired
};
 
export default CellHolder;
