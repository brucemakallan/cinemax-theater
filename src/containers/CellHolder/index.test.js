import React from 'react';
import { mount } from 'enzyme';
import CellHolder from '.';

describe('Test suite for <CellHolder />', () => {
	let props;

	beforeEach(() => {
		props = {
			value: '*',
			type: 'normal',
			id: 'A1'
		};
	});

	it('should render correctly', () => {
		const wrapper = mount(<CellHolder {...props} />);
		expect(wrapper).toMatchSnapshot();
	});
});
