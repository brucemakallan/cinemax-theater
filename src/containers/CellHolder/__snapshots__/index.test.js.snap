// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for <CellHolder /> should render correctly 1`] = `
<CellHolder
  id="A1"
  type="normal"
  value="*"
>
  <Cell
    id="A1"
    onClick={[Function]}
    type="normal"
    value="*"
  >
    <div
      className="cell"
    >
      <div
        className="cellValue normal"
      >
        <button
          id="A1"
          onClick={[Function]}
          title="normal"
          type="button"
        >
          <i
            className="fas fa-chair"
          />
        </button>
      </div>
    </div>
  </Cell>
</CellHolder>
`;
